plugins/workspace-backpack/src/index.js

```diff
@@ -438,43 +439,48 @@ export class Backpack {
    *     cleaned of all unnecessary attributes.
    */
   addItem(item) {
-    if (this.contents_.indexOf(item) !== -1) {
-      return;
-    }
+    this.addItems([item]);
+  }
 
-    this.contents_.unshift(item);
-    while (this.contents_.length > this.maxItems_) {
-      this.contents_.pop();
-    }
-    // TODO: Fire UI event for Backpack content change.
+  /**
+   * Adds multiple items to the backpack.
+   * @param {!Array<string>} items The backpack contents to add.
+   */
+  addItems(items) {
+    const filteredItems = items.filter((item) => {
+      return this.contents_.indexOf(item) === -1;
+    });
+    this.contents_.unshift(...filteredItems);
+    this.trimContents_();
+    Blockly.Events.fire(new BackpackChange(this.workspace_.id));
   }
 
   /**
    * Sets backpack contents XML.
    * @param {!Array<string>} contents The new backpack contents.
    */
   setContents(contents) {
-    this.contents_ = [...contents];
-    while (this.contents_.length > this.maxItems_) {
-      this.contents_.pop();
-    }
-    // TODO: Fire UI event for Backpack content change.
+    this.contents_ = contents.filter((item) => {
+      return this.contents_.indexOf(item) === -1;
+    });
+    this.trimContents_();
+    Blockly.Events.fire(new BackpackChange(this.workspace_.id));
   }
 
   /**
-   * Merges backpack contents XML.
-   * @param {!Array<string>} contents The backpack contents to merge.
+   * Trims the internal contents array so that it does not exceed max size.
+   * @private
    */
-  mergeContents(contents) {
-    contents.forEach((item) => {
-      this.addItem(item);
-    });
+  trimContents_() {
```

Comment:
This feels like something that should be an error rather than silently dropping items, but I'm not sure what UX you have in mind for adding and managing items so I could be convinced otherwise

Reply:
This is how we handle items in the trashcan (silently dropping them), so I followed the same pattern here.
There are other ways we could handle this, we can discuss offline and possibly just remove the maxItem feature for now (and add back later).

Reply:
As discussed offline. I've removed maxItems in this PR and we can address how handle max items in future.